app/main/services/search_service.py

```diff
@@ -134,6 +134,17 @@ def core_search_and_aggregate(index_name, doc_type, query_args, search=False, ag
 
         results = convert_es_results(mapping, res, query_args)
 
+        # WORKAROUND: In Elasticsearch 6 the default highlighter will only return the first sentence if the
+        # highlighter finds no terms to mark. This hack basically makes sure that the always get the full
+        # service description in this case. See https://github.com/elastic/elasticsearch/issues/41066.
+        # Should be fixed in > v6.7.2 :fingers_crossed:.
+        for document in results["documents"]:
+            if "highlight" not in document:
+                break
+            if len(document["highlight"]["serviceDescription"][0]) < len(document["serviceDescription"]):
+                escaped_description = escape(document["serviceDescription"])
+                document["highlight"]["serviceDescription"] = [escaped_description]
+
```

Comment:
So this won't escape our `<mark class='search-result-highlighted-text'>` because anything that _has_ been marked will return the right amount of description and therefore `len(document["highlight"]["serviceDescription"][0]) < len(document["serviceDescription"])` will be false?

Reply:
That's the theory; I think the tests show that it works. Hopefully we can take it out soon though.